app/models/tour_list.rb

```diff
@@ -0,0 +1,39 @@
+class TourList
+  attr_reader :contents
+
+  def initialize(initial_contents)
+    @contents = initial_contents || Hash.new(0)
+    @contents.default = 0
```

Comment:
```suggestion
```
`Hash.new(0)` already does this -- redundant.

Reply:
`@contents.default = 0` is needed when dealing with sessions. I don't know why but if I make the changes you've suggested, everything fails.